src/basic/terminal-util.c

```diff
@@ -857,6 +857,36 @@ unsigned lines(void) {
         return cached_lines;
 }
 
+int terminal_set_size_fd(int fd, unsigned rows, unsigned cols) {
+        struct winsize ws;
```

Comment:
so you use rows=0/cols=0 as magic hint for "don#t change this one". I wonder if it would be better to use `(unsigned) -1` (i.e. UINT_MAX) for this instead, for the simple reason that these settings can actually be set to 0 too, and often are, after all the kernel does auto-reset things to 0/0 on its own anyway. OTOH UINT_MAX is something definitely never seen (for the simple reason that the accepted range ends add USHRT_MAX already). (It's also a tiny bit more similar to how `chown()` does it, where `(uid_t) -1` is how you say "don't change this").